site/source/tutorials/10XX_development_environment/index.md

```diff
@@ -0,0 +1,146 @@
+      "pattern": [
+        {
+          "regexp": "^(\\S+) in (.*)",
+          "severity": 1,
+          "file": 2
+        }, {
+          "regexp": "\\((\\d+),(\\d+)\\):(.*)",
+          "line": 1,
+          "column": 2,
+          "message": 3
+        }
+      ]
+      }
+    }
+  ]
+}
+```
+
+Other tasks to consider adding could be `test` or `watch`. Once configured, the build command (<kbd>⌘⇧B</kbd>/<kbd>Ctrl⇧B</kbd>) will be configured as well as you will have the other
+tasks available in the IDE:
+
+<p class="center">![dropdown of vs code tasks](./resources/run_task.png)</p>
+
+{% section %}
+
+## Debugging
+
+If you are developing a Dojo 2 application in TypeScript, your application code is transpiled from TypeScript to JavaScript. If you are also using dojo build, then your code is also being bundled and minimized. This could make it challenging to debug an application, but we have tried to make Dojo 2 applications integrate well into modern debugging tools.
+
+Source maps are the way on the web we can describe code that has been transformed in some fashion. The dojo build maps the code throughout the process so that both the original TypeScript code and CSS code is available when debugging. This should allow you to set breakpoints and watches on the original code as well as see the original code when there is a run-time error.
+
+{% task 'Debug test app in Chrome Dev Tools' %}
+
+{% instruction 'Create a breakpoint in WorkerContainer' %}
+
+To demonstrate the source mapping between compiled Javascript code and the source TypeScript, you can open the `Sources` tab in dev tools, and browse to `WorkerContainer.ts`. Insert a breakpoint where desired, and upon reloading you should see it pause script execution:
+
+<p class="center">![breakpoint set in chrome dev tools](./resources/breakpoint.png)</p>
+
+{% task 'Install an accessibility inspector' %}
+
+Dojo 2 widgets are designed to be accessible by default, but a full app is much more than the sum of its parts. A good in-browser a11y inspector helps integrate accessibility into the development process. Chrome provides the [Accessibility Developer Tools](ttps://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb), which can be used when inspecting the DOM:
+
+<p class="center">![chrome accessibility inspector](./resources/a11y_inspection.png)</p>
+
+This exposes text and semantics visible to a screen reader, exposing a page model referred to as the [Accessibility Tree](https://developers.google.com/web/fundamentals/accessibility/semantics-builtin/the-accessibility-tree).
+
+Chrome Accessibility Developer Tools does not run an audit against your code, or validate it in any way. For easy in-browser a11y validation, aXe has free open-source extensions for [Chrome](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) and [Firefox](https://addons.mozilla.org/en-us/firefox/addon/axe-devtools/).
```

Comment:
Do we want to say anything about Intern and its support for a11y testing and how Intern is what Dojo 2 uses for its internal testing?

Reply:
I wanted to avoid overlap with the testing tutorial (or was it developer tools?) which is where I imagined Intern + a11y testing would more naturally fit. If you think this tutorial should include a separate block on testing I'd be happy to add it.

Reply:
That's reasonable, it's all a blur to me at times with what should go where. :)